metadetect/test.py

```diff
@@ -244,89 +246,165 @@ def _make_psf(self):
+        print("trial: %d/%d" % (trial+1, ntrial))
 
         mbobs = sim.get_mbobs()
-        mer=detect.MEDSifier(
+        mer = detect.MEDSifier(
             mbobs,
             config['sx'],
             config['meds'],
         )
 
-        mbm=mer.get_multiband_meds()
+        mbm = mer.get_multiband_meds()
 
-        nobj=mbm.size
-        print("        found",nobj,"objects")
+        nobj = mbm.size
+        print("        found", nobj, "objects")
         nobj_meas += nobj
 
         if show:
             _show_mbobs(mer)
             if ntrial > 1 and trial != (ntrial-1):
-                if 'q'==input("hit a key: "):
+                if 'q' == input("hit a key: "):
                     return
 
+    total_time = time.time()-tm0
+    print("time per group:", total_time/ntrial)
+    print("time per object:", total_time/nobj_meas)
 
-    total_time=time.time()-tm0
-    print("time per group:",total_time/ntrial)
-    print("time per object:",total_time/nobj_meas)
 
 def test_metadetect(ntrial=1):
     """
     test full metadetection
     """
 
-    rng=np.random.RandomState()
+    rng = np.random.RandomState()
 
     tm0 = time.time()
-    nobj_meas = 0
 
     sim = Sim(rng)
-    config={}
+    config = {}
     config.update(TEST_METADETECT_CONFIG)
 
     for trial in range(ntrial):
-        print("trial: %d/%d" % (trial+1,ntrial))
+        print("trial: %d/%d" % (trial+1, ntrial))
 
         mbobs = sim.get_mbobs()
-        res = metadetect.do_metadetect(config, mbobs, rng)
+        metadetect.do_metadetect(config, mbobs, rng)
+
+    total_time = time.time()-tm0
+    print("time per:", total_time/ntrial)
+
+
+def test_shear_pos(show=False):
+    """
+    test shearing and unshearing positions
+    """
+    from . import shearpos
+
+    step = 0.10
+
+    dims = 100, 100
+    im = np.zeros(dims)
+    cen = (np.array(dims) - 1)/2
+    jacobian = ngmix.Jacobian(
+        row=cen[0],
+        col=cen[1],
+        dudrow=0.263,
+        dudcol=-0.01,
+        dvdrow=+0.01,
+        dvdcol=0.263,
+    )
+
+    obs = ngmix.Observation(
+        im,
+        jacobian=jacobian,
+    )
+
+    shears = ['noshear', '1p', '1m', '2p', '2m']
+
+    rows, cols = np.mgrid[
+        :dims[0]:5,
+        :dims[1]:5,
+    ]
+    rows = rows.ravel()
+    cols = cols.ravel()
+
+    for sstr in shears:
+        srows, scols = shearpos.shear_positions(
+            rows,
+            cols,
+            sstr,
+            obs,
+            step=step,
+        )
+
+        if show:
+            _show_pos(rows, cols, srows, scols, title=sstr)
+
+        crows, ccols = shearpos.unshear_positions(
+            srows,
+            scols,
+            sstr,
+            obs,
+            step=step,
+        )
+
+        if show:
+            _show_pos(rows, cols, crows, ccols, title=sstr+' unsheared')
+
+        assert np.allclose(rows, crows), 'checking rows inverse works'
+        assert np.allclose(cols, ccols), 'checking rows inverse works'
```

Comment:
Ideally, a test like this would also check the values to make sure they are correct compared to a simple computation.

Reply:
We probably also want to compare to what galsim does to an image with say a single pixel filled in, to make sure everyone is using the same conventions and we are properly unshearing what galsim did. I don't remember if I did that or not.

Reply:
I added this exact test and it passes